datastore.go

```diff
@@ -59,10 +57,29 @@ func (d *datastore) Get(key ds.Key) (value interface{}, err error) {
 	if err != nil {
 		return nil, err
 	}
-	if item.Value() == nil {
+
+	var bytes []byte
+
+	err = item.Value(func(b []byte) error {
```

Comment:
I think we have to make a copy here. This is a direct access of the memmapped file IIUC